Sources/MongoKitten/AggregateStage.swift

```diff
@@ -82,6 +101,44 @@ public struct AggregateBuilderStage {
         ])
     }
     
+	/// The `lookup` aggregation performs a join from another collection in the same database. This aggregation will add a new array to
+	/// your document including the matching documents.
+	///
+	/// # MongoDB-Documentation:
+	/// [Link to the MongoDB-Documentation](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/)
+	///
+	/// # Example:
+	/// There are two collections, named `users` and `userCategories`. In the `users` collection there is a reference to the _id
+	/// of the `userCategories`, because every user belongs to a category.
+	///
+	/// If you now want to aggregate all users and the corresponding user category, you can use the `$lookup` like this:
+	///
+	/// ```
+	/// let pipeline = userCollection.aggregate([
+	///     .lookup(from: "userCategories", "localField": "categoryID", "foreignField": "_id", newName: "userCategory")
+	/// ])
+	/// 
+	/// pipeline.execute().whenComplete { result in
+	///    ...
+	/// }
+	/// ```
+	///
+	/// # Hint:
+	/// Because the matched documents will be inserted as an array no matter if there is only one item or more, you may want to unwind the joined documents:
+	///
+	/// ```
+	/// let pipeline = myCollection.aggregate([
+	///     .lookup(from: ..., newName: "newName"),
+	///     .unwind(fieldPath: "$newName")
+	/// ])
+	/// ```
+	///
+	/// - Parameters:
+	///   - from: the foreign collection, where the documents will be looked up
+	///   - localField: the name of the field in the input collection that shall match the `foreignField` in the `from` collection
+	///   - foreignField: the name of the field in the `fromCollection` that shall match the `localField` in the input collection
+	///   - newName: the collecting matches will be inserted as an array to the input collection, named as `newName`
+	/// - Returns: returns an `AggregateBuilderStage`
```

Comment:
Is the first `returns` word in `return an AggregateBuilderStage` added for readability in Xcode? It seems redundant in the `Returns` section.

Reply:
That's right. Should be just like this?

/// - Returns: An `AggregateBuilderStage`

Reply:
If you option-click on the function definition or any call to it, your best guess should guide you :)